miner/miner.go

```diff
@@ -33,6 +35,14 @@ var log = logging.Logger("miner")
 // returns a callback reporting whether we mined a blocks in this round
 type waitFunc func(ctx context.Context, baseTime uint64) (func(bool, error), error)
 
+func randTimeOffset(width time.Duration) time.Duration {
+	buf := make([]byte, 8)
+	rand.Reader.Read(buf)
+	val := time.Duration(binary.BigEndian.Uint64(buf) % uint64(width))
+
+	return val - (width / 2)
```

Comment:
Why `width/2`, we always move the boundary back.

Reply:
because I want it to be framed as 6s +/- 500ms

Reply:
ahh, I misread it :/